src/commands/admin/admin-subcommands/blacklist.ts

```diff
@@ -0,0 +1,216 @@
+
+        // Should ideally not be null but in the off-chance,
+        // throw an error
+        if (!bl) {
+            await createBlacklist();
+        }
+
+        const users: Map<string, string> = bl.users;
+
+        // Do nothing if the user has already been blacklisted
+        if (users.get(uid)) {
+            await ctx.followUp("The specified user is already blacklisted");
+            return;
+        }
+
+        const millisToSecs: number = 1000;
+
+        // Ask for confirmation before blacklisting the target user
+        const embed: EmbedBuilder = new EmbedBuilder()
+            .setTitle(`Blacklist ${user.username}?`)
+            .setDescription("The following action will be difficult to reverse, are you sure?")
+            .setColor(RED)
+            .setAuthor({ name: "Blacklist", iconURL: ctx.client.user.avatarURL() })
+            .setThumbnail(user.avatarURL())
+            .addFields(
+                { name: "User ID", value: user.id, inline: true },
+                { name: "Created", value: `<t:${Math.floor(user.createdTimestamp / millisToSecs)}>`, inline: true },
+                { name: "Reason", value: reason }
+            );
+        
+        const yes: ButtonBuilder = new ButtonBuilder()
+            .setCustomId("yes")
+            .setLabel("Yes")
+            .setStyle(ButtonStyle.Success);
+        
+        const no: ButtonBuilder = new ButtonBuilder()
+            .setCustomId("no")
+            .setLabel("No")
+            .setStyle(ButtonStyle.Danger);
+
+        const row: ActionRowBuilder<ButtonBuilder> = new ActionRowBuilder<ButtonBuilder>()
+            .addComponents(yes, no);
+
+        const prompt: Message = await ctx.followUp({
+            embeds: [embed],
+            components: [row]
+        });
+
+        // Check if the user pressing the button is the same user
+        const check: CollectorFilter<[ButtonInteraction]> = (m: ButtonInteraction): boolean => m.user.id === ctx.user.id;
+
+        // Disable the buttons to signify that the interaction has ended
+        row.components.forEach(c => c.setDisabled(true));
+
+        try {
+            const confirmation: MessageComponentInteraction = await prompt.awaitMessageComponent({ filter: check, time: 10_000 });
+
+            // End the interaction as soon as a button is pressed
+            await confirmation.update({
+                components: [row]
+            });
+
+            // If the user selected no, abort blacklisting
+            if (confirmation.customId === "no") {
+                await ctx.followUp("Blacklisting denied, aborting opertation");
+                return;
+            }
+        } catch (e) {
+            // End the interaction upon timeout
+            await prompt.edit({
+                components: [row]
+            });
+            // If the interaction timed out, abort as well
+            await ctx.followUp("Interaction timed out, aborting operation");
+            return;
+        }
```

Comment:
If the administrator changes their mind or times out, the operation will be aborted